logistics_project/apps/malawi/handlers/abstract/base.py

```diff
@@ -14,4 +15,24 @@ def respond(self, template, **kwargs):
     
     @property
     def responded(self):
-        return self._responded
\ No newline at end of file
+        return self._responded
+
+
+class FacilityUserHandler(KeywordHandler):
+    """
+    Intended to be used when Facility users need to interact with the system
+    over SMS.
+    """
+
+    def validate_contact(self):
+        """
+        Checks if the contact who sent this message can use the keyword
+        and returns that contact's supply point if so.
+        """
+        contact = getattr(self.msg,'logistics_contact', None)
```

Comment:
you can add `@logistics_contact_required` to the `handle` function ensure it is always set (and respond if not)